app/src/main/java/com/artemchep/basics_multithreading/MainActivity.java

```diff
@@ -54,11 +55,24 @@ public void insert(final WithMillis<Message> message) {
         mList.add(message);
         mAdapter.notifyItemInserted(mList.size() - 1);
 
+        CipherThread task = new CipherThread(message.value.plainText);
+        task.setCICallback(new CipherI() {
+            @Override
+            public void updateUICallback(String cypheredText, long executionTime) {
+                final Message messageNew = message.value.copy(cypheredText);
+                final WithMillis<Message> messageNewWithMillis = new WithMillis<>(messageNew, (executionTime) / 10000000);
+                update(messageNewWithMillis);
+            }
+        });
+        Thread t = new Thread(task);
+        t.start();
```

Comment:
Should not have commented out the welcome message, as it states that all messages should be processed sequentially.